### ISSUE TITLE
Private Production Build

### ISSUE THREAD

**Issue by CharithaDhananjaya** [freeCodeCamp/freeCodeCamp#38580]
I'm trying to deploy a private instance of FCC for my students. But I couldn't do it somehow. Creating an account and Sign in to the private instance is not working with development user build. So if someone know how to build the private instance as a production build, please let me know your guid, in order to make this successful. Thanks in advance.

**Comment by ojeytonwilliams** [comment]
@davidreke if memory serves, you set up fcc for a bootcamp, do you have any notes on how you did that?

@CharithaDhananjaya could you outline what you've tried? There's a few things that could be going wrong.

**Comment by CharithaDhananjaya** [comment]
@ojeytonwilliams What I did was, 
1.) get downloaded the code base from GitHub.
2.) then I follow the Local Deployment steps mentioned in the docs (https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/how-to-setup-freecodecamp-locally.md) 
3.) just did like this, 
i. npm ci 
ii. npm run seed
iii. npm run develop
--- then I could access the development user fcc from port 8000.

**Comment by CharithaDhananjaya** [comment]
that works seems okay, but for my need it was not the requirement. So I'll be really happy if someone can help on that.

**Comment by raisedadead** [comment]
Hi,

Thanks for your query. 

TL;DR: You are unable to white-label the stack OOTB. We do not support it, because it's not a requirement for us.

But are able to answer specific questions on the codebase, architecture.

Just to give you someplace to start and get back to us with meaningful well researched questions:

1. We use Loopback (v3) as our API/ORM framework, it supports Passport.js (because Loopback internally uses express).
2. You should be able to install any passport adapter for your use-case and modify the `User-Identity` object to parse the email (which is what Loopback uses for the "Username" equivalent).
3. Then on you should be able to deploy the `api-server` and the `client` separately in a production like environment.
4. We use PM2 to serve the API. The client is a static Gatsby app.
5. You may need to tweak the code for your specific use-case, and handle CORS, etc.

**Comment by CharithaDhananjaya** [comment]
Thank, I got bit cleared. My intention is not to white labelling or nay changes. I just want FCC as it is but locally, Since my students are below 13 years and just want to develop a coding camp with their parents involving. So if this local system successful I can develop a dashboard for parents using the api, not mixing with this global platform. 

Thanks for you valuable information, I'll consider the points you mentioned. 
Still expecting if there are someone who already done this, for quick deployment locally.

**Comment by raisedadead** [comment]
Sure thing, and just clear any confusion we are more than happy to add changes (via pull-requests) to the codebase (and answer any queries about archi), for supporting this OOTB.

**Comment by CharithaDhananjaya** [comment]
Are there any documentation on production build, like I just stuck in moving from development user to production build.

**Comment by raisedadead** [comment]
There is some in our DevOps guide, but it is not remotely close to everything you need. You would need to remove the `LOCAL_MOCK_AUTH` from your keys and setup Auth0. ( And a bunch of unnecessary things like Stripe, PayPal, etc.)

The production setup will not solve your requirement for users below age of 13. You need a local stack, which means some decoupling of the boot scripts (in API) and adding adapters for passport.

If you want to go that route, you will be needing to explore the code from the pointers in my previous comment.

Goodluck.

**Comment by CharithaDhananjaya** [comment]
Okay thanks, I got much cleared. If no issue, let me know the specific Gatsby build command for the production build. 
Thank in advance.
